community/saas-companies-engagement-procedure.md

```diff
@@ -0,0 +1,118 @@
+## Procedure for Engaging With SaaS Companies
+
+### Summary
+
+This document outlines the procedure for engaging with SaaS companies, and
+how SaaS companies can represent themselves and their interests in the SPIFFE
+community.
+
+The goal is to provide a clear and consistent process for engaging with 
+SaaS companies that are included but not limited to those that use SPIFFE and 
+SPIRE, provide a proprietary SPIFFE-compatible service, or are interested in 
+contributing to the SPIFFE project.
+
+More specifically, this document includes guidelines for communication, 
+technical contact designation, and handling potential disruptive participants.
+
+### Technical Contacts
+
+#### Objective
+
+To ensure effective communication and support for SaaS consumers 
+using the SPIFFE specification by designating responsible technical contacts.
+
+#### Guidelines
+
+##### Designation of Contacts
+
+Each SaaS company that directs customers to the SPIFFE Slack channel must 
+designate two technical contacts.
+
+These contacts will serve as the primary point of contact for addressing queries 
+and facilitating communication between the customers and the Spiffy community.
+
+##### Documentation of Contacts
+
+A dedicated folder for each company will be created in the SPIFFE repository
+under the `./community/saas-technical-contacts` directory.
+
+The folder will contain:
+
+* Names and contact information of the designated technical contacts.
+* Guidelines for interaction and support procedures.
```

Comment:
It's not clear to me what "Guidelines for interaction" actually involves. Is this simply a case of pointing folks towards the appropriate ticket portal on the SAAS provider side?

Reply:
I believe it will be up to SSC and Company to agree upon. 

These could include steps for submitting tickets, expected response times, preferred communication channels, and any other relevant interaction policies.

Thanks for the feedback, I’ll clarify that part a bit.

Reply:
@strideynet (regarding the above longer comment)

Thank you for sharing your thoughts and concerns. I understand and appreciate the challenges you've highlighted regarding vendors' responsibility in moderating community behavior, especially when it involves individuals without a direct commercial relationship.

Our aim with these guidelines is to ensure a positive and productive community environment. However, we also do recognize the difficulties of moderation and the potential burden it may place on technical contacts.

Humans are nondeterministic beings. And that makes all human-related issues convoluted.

Given the complexities you've mentioned, discussing this further within the SPIFFE Steering Committee (SSC) would be beneficial. We want to ensure that our policies are practical and do not discourage valuable contributions to the community.

Your feedback is invaluable as we continue to refine our guidelines.

Thanks again all your comments and feedback.

Reply:
> Humans are nondeterministic beings. And that makes all human-related issues convoluted.

Never a truer thing said :")

I appreciate you taking the feedback into account - I'll admit that I've not had much of an opportunity to give back to the SPIFFE community yet but it's certainly in my plans for this year.

---

I think if I were to expand a little on my thoughts - based on this document, it would appear that there's two broad goals with the proposed vendor engagement strategy:

- Technical: Setting out ways for the SPIFFE team to get in contact with technical folks at these vendor projects, for the purposes of technical discussion etc.
- Social: Setting out a level of commitment that that vendor needs to make to fostering a positive and growing SPIFFE community, and engaging in the work of trying to moderate it.

Both of these are good goals and I think it's a reasonable expectation of vendor to "give back" in both these ways.

However, I'm not sure bundling the second element with the first necessarily makes sense. Perhaps some of the time it does - I'd be happy to engage with both elements on behalf of Teleport - but when it comes to picking the "second" contact, I face a choice:

- Do I ask someone who is much stronger at community engagement and moderation, but weaker when it comes to the technical details of our SPIFFE implementation? (e.g a member of our devrel team)
- Do I ask someone who is much stronger at the technical details of our SPIFFE implementation but weaker when it comes to community engagement? (e.g a member of my engineering team)

I don't think this "trade off" is specific to us, I've seen plenty of similar orgs where the skills needed here are broken up into different parts of the organisation. 

What I have in mind is perhaps breaking up these roles:

- Technical Contacts
  - Intended to be individuals with a high level of context on the implementation of the vendor's SPIFFE product.
  - Role: contact point for SPIFFE contributors who have questions about the details of a vendor implementation, and for escalating technical queries.
- Community Contacts
  - Intended to be individuals who perform devrel/community work for the vendor.
  - Role: Engage with moderation side.
  - Role: good contact point for discussing content which could benefit both the vendor and the SPIFFE project as a whole (e.g publicising case studies, use case statements, blog posts)

I think it wouldn't be a problem to have an individual be both a Technical Contact and a Community Contact but by breaking up the roles we can make this more flexible and potentially more accessible. I can imagine individuals who'd be keen to be more involved with the relationship with SPIFFE, but might be hesitant to take on both technical and social aspects.